component/loki/process/process.go

```diff
@@ -131,7 +131,7 @@ func (c *Component) handleIn(ctx context.Context, wg *sync.WaitGroup) {
 			select {
 			case <-ctx.Done():
 				return
-			case c.processIn <- entry:
+			case c.processIn <- entry.Clone():
```

Comment:
Does loki.process usually modify entries? If it does, this is fine. But if modifying entries is rare, we should add a TODO here to switch to copy-on-write as an optimization to avoid unnecessary allocations.

Reply:
IMO at some point we should revisit how logs are passed along but this feels totally reasonable for now.